dotnet/src/Microsoft.AutoGen/Core/InProcessRuntime.cs

```diff
@@ -274,6 +274,7 @@ public ValueTask StartAsync(CancellationToken token = default)
         }
 
         this.shutdownSource = new CancellationTokenSource();
+        this.shouldContinue = () => true;
```

Comment:
what is the difference here between a lambda that evaluates to true and a value that is true?

Reply:
This enables a callback to be inserted as in the case of RunUntilIdle(); but when we restart, we need to change it back so it does not immediately stop the message pumping loop because the message queue starts out empty.

Reply:
I changed this to be reset during the use of RunUntilIdle